sklearn/preprocessing/discretization.py

```diff
@@ -166,7 +166,7 @@ def _validate_n_bins(self, n_features, ignored):
         """
         orig_bins = self.n_bins
         if isinstance(orig_bins, numbers.Number):
-            if not isinstance(orig_bins, np.int):
+            if not isinstance(orig_bins, (numbers.Integral, np.integer)):
```

Comment:
just note that it will eventually be changed to something like SCALAR_INTEGER_TYPES in #10017